File: Proyecto/frontend/nbaAnalitics_fr/src/app/models/playerStats.ts
export class PlayerStats{
  id:number;
	player_id:number;
	player_name:string;
	nickname:string;
	team_id:number;
	team_abbrev:string;
	age:number;
	gp:number;
	won:number;
	lost:number;
	won_PCT:number;
	fgm:number;
	fga:number;
	fg_PCT:number;
	fg3M:number;
	fg3A:number;
	fg3_PCT:number;
	ftm:number;
	fta:number;
	ft_PCT:number;
	oreb:number;
	dreb:number;
	reb:number;
	ast:number;
	tov:number;
	stl:number;
	blk:number;
	blka:number;
	pf:number;
	pts:number;
  min:number;

  constructor(id:number = 0,
    player_id:number = 0,
    player_name:string = '',
    nickname:string = '',
    team_id:number = 0,
    team_abbrev:string = '',
    age:number = 0,
    gp:number = 0,
    won:number = 0,
    lost:number = 0,
    won_PCT:number = 0,
    fgm:number = 0,
    fga:number = 0,
    fg_PCT:number = 0,
    fg3M:number = 0,
    fg3A:number = 0,
    fg3_PCT:number = 0,
    ftm:number = 0,
    fta:number = 0,
    ft_PCT:number = 0,
    oreb:number = 0,
    dreb:number = 0,
    reb:number = 0,
    ast:number = 0,
    tov:number = 0,
    stl:number = 0,
    blk:number = 0,
    blka:number = 0,
    pf:number = 0,
    pts:number = 0,
    min:number = 0){
      this.id = id;
      this.player_id = player_id;
      this.player_name = player_name;
      this.nickname = nickname;
      this.team_id = team_id;
      this.team_abbrev = team_abbrev;
      this.age = age;
      this.gp = gp;
      this.won = won;
      this.lost = lost;
      this.won_PCT = won_PCT;
      this.fgm = fgm;
      this.fga = fga;
      this.fg_PCT = fg_PCT;
      this.fg3M = fg3M;
      this.fg3A = fg3A;
      this.fg3_PCT = fg3_PCT;
      this.ftm = ftm;
      this.fta = fta;
      this.ft_PCT = ft_PCT;
      this.oreb = oreb;
      this.dreb = dreb;
      this.reb = reb;
      this.ast = ast;
      this.tov = tov;
      this.stl = stl;
      this.blk = blk;
      this.blka = blka;
      this.pf = pf;
      this.pts = pts;
      this.min = min;
  }

}

File: Proyecto/backend/demo/src/main/java/com/example/demo/players/model/PlayerStats.java
package com.example.demo.players.model;

import java.util.Objects;

import com.example.demo.commonstats.CommonStats;

public class PlayerStats extends CommonStats{

	private int playerId;
	private String playerName;
	private String nickname;
	private String teamAbbrev;
	private double age;
	private int gp;
	private int blka;
	private int min;
	private int won;
	private int lost;
	private double wonPct;

	public int getPlayerId() {
		return playerId;
	}

	public void setPlayerId(int playerId) {
		this.playerId = playerId;
	}

	public String getPlayerName() {
		return playerName;
	}

	public void setPlayerName(String playerName) {
		this.playerName = playerName;
	}

	public String getNickname() {
		return nickname;
	}

	public void setNickname(String nickname) {
		this.nickname = nickname;
	}

	public String getTeamAbbrev() {
		return teamAbbrev;
	}

	public void setTeamAbbrev(String teamAbbrev) {
		this.teamAbbrev = teamAbbrev;
	}

	public double getAge() {
		return age;
	}

	public void setAge(double age) {
		this.age = age;
	}

	public int getGp() {
		return gp;
	}

	public void setGp(int gp) {
		this.gp = gp;
	}


	public int getBlka() {
		return blka;
	}

	public void setBlka(int blka) {
		this.blka = blka;
	}

	public int getMin() {
		return min;
	}

	public void setMin(int min) {
		this.min = min;
	}
	
	public int getWon() {
		return won;
	}

	public void setWon(int won) {
		this.won = won;
	}

	public int getLost() {
		return lost;
	}

	public void setLost(int lost) {
		this.lost = lost;
	}

	public double getWonPct() {
		return wonPct;
	}

	public void setWonPct(double wonPct) {
		this.wonPct = wonPct;
	}

	@Override
	public int hashCode() {
		return Objects.hash(age, ast, blk, blka, dreb, fg3Pct, fg3a, fg3m, fgPct, fga, fgm, ftPct, fta, ftm, gp, id,
				lost, min, nickname, oreb, pf, playerId, playerName, pts, reb, stl, teamAbbrev, teamId, tov, won,
				wonPct);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		PlayerStats other = (PlayerStats) obj;
		return Double.doubleToLongBits(age) == Double.doubleToLongBits(other.age) && ast == other.ast
				&& blk == other.blk && blka == other.blka && dreb == other.dreb
				&& Double.doubleToLongBits(fg3Pct) == Double.doubleToLongBits(other.fg3Pct) && fg3a == other.fg3a
				&& fg3m == other.fg3m && Double.doubleToLongBits(fgPct) == Double.doubleToLongBits(other.fgPct)
				&& fga == other.fga && fgm == other.fgm
				&& Double.doubleToLongBits(ftPct) == Double.doubleToLongBits(other.ftPct) && fta == other.fta
				&& ftm == other.ftm && gp == other.gp && id == other.id && lost == other.lost && min == other.min
				&& Objects.equals(nickname, other.nickname) && oreb == other.oreb && pf == other.pf
				&& playerId == other.playerId && Objects.equals(playerName, other.playerName) && pts == other.pts
				&& reb == other.reb && stl == other.stl && Objects.equals(teamAbbrev, other.teamAbbrev)
				&& teamId == other.teamId && Double.doubleToLongBits(tov) == Double.doubleToLongBits(other.tov)
				&& won == other.won && Double.doubleToLongBits(wonPct) == Double.doubleToLongBits(other.wonPct);
	}

}
